### PR TITLE
Add vault helm chart

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/5615

### CODE REVIEW

**Comment by mazzy89** [comment]
is this chart inspired to any upstream one or you sketched from the ground?

**Comment by corest** [comment]
From the ground completely.

It is tested now on ginger cluster and works fine for me
Some security on whitelisting will be applied later
Also with next PR I'll add circleci orbs to push it into appcatalog

File: helm/vault/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.name }}-secret
+  namespace: {{ .Values.namespace }}
+  labels:
+    giantswarm.io/service-type: "managed"
+type: Opaque
+data:
+  aws_access_key_id: {{ .Values.aws.access_key_id | b64enc }}
```
Comment (mazzy89) [inline helm/vault/templates/secret.yaml]: Don't we have anything like `kube2iam` instead to have keys here?

Comment (corest) [reply helm/vault/templates/secret.yaml]: What do you mean by ‘Don’t we have?’
It is possible to use kube2iam. However you need to setup assuming role for workers. And that solution is not that effective, as having api creds, which allows three operations over sinngle kms key. So in this case using kube2iam is a huge overhead. Even we would have at least one more service, which requires aws, then we could think about kube2iam

Comment (mazzy89) [reply helm/vault/templates/secret.yaml]: Make sense. 👍

**Pull request merged**